File: apis/elysium-chat-api/src/main/java/com/seventh_root/elysium/api/chat/ElysiumChatGroup.java
package com.seventh_root.elysium.api.chat;

import com.seventh_root.elysium.api.player.ElysiumPlayer;
import com.seventh_root.elysium.core.database.TableRow;

import java.util.Collection;

public interface ElysiumChatGroup extends TableRow {

    String getName();

    void setName(String name);

    Collection<? extends ElysiumPlayer> getPlayers();

    void addPlayer(ElysiumPlayer player);

    void removePlayer(ElysiumPlayer player);

    Collection<? extends ElysiumPlayer> getInvited();

    void invite(ElysiumPlayer player);

    void uninvite(ElysiumPlayer player);

    void sendMessage(ElysiumPlayer sender, String message);

}

File: apis/elysium-chat-api/src/main/kotlin/com/seventh_root/elysium/api/chat/ElysiumChatGroup.kt
package com.seventh_root.elysium.api.chat

import com.seventh_root.elysium.api.player.ElysiumPlayer
import com.seventh_root.elysium.core.database.TableRow

interface ElysiumChatGroup : TableRow {

    var name: String
    val players: Collection<ElysiumPlayer>
    fun addPlayer(player: ElysiumPlayer)
    fun removePlayer(player: ElysiumPlayer)
    val invited: Collection<ElysiumPlayer>
    fun invite(player: ElysiumPlayer)
    fun uninvite(player: ElysiumPlayer)
    fun sendMessage(sender: ElysiumPlayer, message: String)

}
